### ISSUE TITLE
Need RETRY in `congo` session, if something went wrong and disconnected the ssh session

### ISSUE THREAD

**Issue by neo954** [xcat2/goconserver#1]
This bug is against `goconserver` `Version: 0.1, BuildTime: 2017-11-02T06:57:35-0400 Commit: 7e9278e88eb4f3035707b41f0a717fede7291498` running on a ppc64le Red Hat 7.4 Linux environment.

**The recreation steps**

Create a node definition in `congo`. And then connect to the newly created node. BTW, the node, `c910f03c01p09` is just another common Linux node, which runs Red Hat 7.4.

```
# congo create c910f03c01p09 driver=ssh ondemand=false \
> --params user=root,host=10.3.1.9,port=22,password=a_password
# congo console c910f03c01p09
```

While the console session is running, on the node `c910f03c01p09`, `kill` the `sshd` process which serve the connection, with a `KILL` signal.

```
[root@c910f03c01p09 ~]# ps axf
  PID TTY      STAT   TIME COMMAND
<<<...omit...>>>
18489 ?        Ss     0:00 /usr/sbin/sshd -D
18964 ?        Ss     0:00  \_ sshd: root@pts/0
18966 pts/0    Ss+    0:00      \_ -bash
<<<...omit...>>>
[root@c910f03c01p09 ~]# kill -KILL 18964
```

And then, back to the `congo` session`, the session get `EOF` on both directions and was dropped.

```
# congo console c910f03c01p09
goconserver(2017-11-07 03:20:19): Hello 127.0.0.1:57208, welcome to the session of c910f03c01p09

[root@c910f03c01p09 ~]# date
Tue Nov  7 03:20:26 EST 2017
[root@c910f03c01p09 ~]# EOF
                           EOF
```

**What actually happened**

From the end user point of view, the `congo` session was dropped when network problem occurred, or remote server reached problem.

**What is expected**

* While a network problem occur,  or the remote server reach problem and disconnect the ssh session, `goconserver` should try to reconnect the ssh session behind the screen, and maintain the `congo` session.
* If the network problem last for a quite long period of time, use binary exponential backoff might be better.

**Comment by chenglch** [comment]
The fix for this issue has been included in the commit 906cd810f009dd1b0b22a75ea0c4efc1cbb2e840

**Comment by neo954** [comment]
This issue was fixed long time ago. I will close this one.
